### PR TITLE
Make mypy error checking opt-out instead of opt-in

### PR DESCRIPTION
**Pull request by Carreau**
Currently mypy configuration is opt-in, where we ignore errors by default.

This PR implements the opposite: when new code / new files are added to the repo they are checked by mypy. Existing known issues are still ignored, but we can progressively work our way down the outstanding files, while not adding new typing errors.

 #2751

### CODE REVIEW

**Review by Czaki — APPROVED** [review]
It will be nice to check if we already could make this list shorter

**Comment by Carreau** [comment]
> It will be nice to check if we already could make this list shorter

That's already the shortest list. 
I've remove all files that are passing mypy, or technically added only files failing.

I have [an upcoming PR](https://github.com/Carreau/napari/compare/mypy-ignore-explicit...Carreau:napari:more-typing?expand=1) that already remove a few of the file here because of trivial typing changes, but did not wanted to mix multiple things.

**Comment by Czaki** [comment]
So I will merge this PR tomorrow.

File: pyproject.toml [hunk 1]
```diff
@@ -216,3 +220,229 @@ check_untyped_defs = true
 # disallow_any_generics = true
 # no_implicit_reexport = true
 # disallow_untyped_defs = true
+
+
+# individual ignore error
+# we should strive to remove those
```
Comment (andy-sweet) [inline pyproject.toml:226]: Might be good to reference the issue tracking that effort: https://github.com/napari/napari/issues/2751

Comment (Carreau) [reply pyproject.toml:226]:
```suggestion
# we should strive to remove those
# See https://github.com/napari/napari/issues/2751
```

**Comment by Carreau** [comment]
> That's already the shortest list.

Funny thing, as I'm adding types more files start to fail so the list may grow at some point before becoming shorter 😢.

**Comment by dstansby** [comment]
Is there an easy/automated way to autogenerate this list? Would be helpful when I'm going through the typing fix PRs I've been opening.

**Comment by dstansby** [comment]
Actually, I guess the easier workflow is just delete one of the items in the new list, and then fix the new errors, so autogeneration probably not that useful.

**Comment by Carreau** [comment]
You can remove the ignore_errors and regererate with 

```bash
$ mypy napari | cut -f1 -d: | sort | uniq  | tr '/' '.'  | sed 's/\.py//'
```

Roughtly.

**Comment by psobolewskiPhD** [comment]
@Carreau Could you edit the PR description to be more explicit for purposes of the merge commit?

Instead of:
> This seems like ...

Go with something like:
> This PR implements the opposite: when new code / new files are added to the repo they are checked by mypy. Existing known issues are still ignored, but we can progressively work our way down the outstanding files, while not adding new typing errors.

(If I'm understanding correctly anyways)

Thanks!

**Comment by Carreau** [comment]
> Could you edit the PR description to be more explicit for purposes of the merge commit

Sure, done.

**Comment by Carreau** [comment]
thanks!

**Pull request merged**